module/VuFind/src/VuFind/ILS/Driver/XCNCIP2.php

```diff
@@ -509,31 +557,45 @@ protected function getHoldingsForChunk($current, $aggregateId = null,
         );
         $volume = (string)($volume[0] ?? '');
 
+        $dateDue = $current->xpath(
+            'ns1:DateDue' .
+            '| ' .
+            'ns1:ItemOptionalFields/ns1:DateDue'
+        );
+        $dateDue = !empty($dateDue)
+            ? $this->displayDate((string)$dateDue[0]) : null;
+
+        $isHoldable = $this->isItemHoldable($current);
         // Build return array:
         $return = [
             'id' => $aggregateId,
             'availability' =>  $this->isAvailable($status),
             'status' => $status,
-            'item_id' => (string)($itemId[0] ?? ''),
+            'item_id' => $itemId,
             'bib_id' => $bibId,
-            'item_agency_id' => (string)($itemAgencyId[0] ?? ''),
+            'item_agency_id' => $itemAgencyId,
             'location' => $location,
             'reserve' => 'N',       // not supported
             'callnumber' => $itemCallNo,
-            'duedate' => '',        // not supported
+            'duedate' => $dateDue,
             'volume' => $volume,
             'number' => $number,
             'barcode' => ($itemType === 'Barcode')
-                ? (string)$itemId[0] : 'Unknown barcode',
-            'is_holdable'  => true,
-            'addLink' => true,
+                ? $itemId : 'Unknown barcode',
+            'is_holdable'  => $isHoldable,
+            'addLink' => empty($this->getPatronBlocks($patron))
+                ? $isHoldable : false,
             'holdtype' => $this->getHoldType($status),
             'storageRetrievalRequest' => 'auto',
             'addStorageRetrievalRequestLink' => 'true',
         ];
         if (strtolower($status) === 'circulation status undefined') {
             $return['use_unknown_message'] = true;
         }
+        if (!empty($collection)) {
+            $return['collection_desc']= $collection;
```

Comment:
Missing space in front of = here. Not sure why none of the style checkers complained.

Reply:
Hm, strange. Anyway I fixed it.